csharp/Facebook.Yoga/YogaLogger.cs

```diff
@@ -8,38 +8,42 @@
  */
 
 using System;
-using System.Runtime.InteropServices;
 
 namespace Facebook.Yoga
 {
     internal static class YogaLogger
     {
-        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
         public delegate void Func(YogaLogLevel level, string message);
 
         private static bool _initialized;
-        private static Func _managedLogger = null;
 
         public static Func Logger = null;
 
         public static void Initialize()
         {
             if (!_initialized)
             {
-                _managedLogger = (level, message) => {
-                    if (Logger != null)
-                    {
-                        Logger(level, message);
-                    }
-
-                    if (level == YogaLogLevel.Error)
-                    {
-                        throw new InvalidOperationException(message);
-                    }
-                };
-                Native.YGInteropSetLogger(_managedLogger);
+
+                Native.YGInteropSetLogger(LoggerInternal);
```

Comment:
You need to store the delegate in a static field, otherwise the GC will collect it and the app will crash when the callback is called:

    static Func _logger_internal = LoggerInternal;

    Native.YGInteropSetLogger (_logger_internal);